packages/vitest/src/runtime/mocker.ts

```diff
@@ -73,7 +81,26 @@ export class VitestMocker {
   }
 
   public getSuiteFilepath(): string {
-    return getWorkerState().filepath || 'global'
+    return this.executor.state.filepath || 'global'
+  }
+
+  private getErrorConstructor(): typeof Error {
+    const context = this.executor.options.context
+    if (!context)
+      return Error
+    return vm.runInContext('Error', context)
+  }
+
+  private createError(message: string) {
+    const Error = this.getErrorConstructor()
+    return new Error(message)
+  }
+
+  private getFinalConstructors(): GlobalConstructors {
```

Comment:
Copy paste the same logic from executor (`this.primitives`), remove `getErrorConstructor`